camel/toolkits/arxiv_toolkit.py

```diff
@@ -153,3 +167,25 @@ def get_tools(self) -> List[FunctionTool]:
             FunctionTool(self.search_papers),
             FunctionTool(self.download_papers),
         ]
+
+
+# Helper function to verify if a URL points to a PDF file
+def check_url_validity(url: str) -> bool:
+    r"""Check if a given URL is valid and points to a PDF file.
+
+    Args:
+        url (str): The URL to check.
+
+    Returns:
+        bool: True if the URL is valid and points to a PDF,
+            False otherwise.
+    """
+    try:
+        response = requests.get(url, stream=True, timeout=5)
+        if response.status_code == 200:
+            content_type = response.headers.get('Content-Type', '').lower()
+            return content_type == 'application/pdf'
+    except requests.RequestException:
+        pass
+
+    return False
```

Comment:
i think we don't need to add this check, we can just add try except for `text = arxiv_to_text(paper_info["pdf_url"])`

Reply:
Hi Wendong thanks for the suggestion. I thought the arxiv2text would have some delay between receiving the erroneous URL and raising an error and it would be better to check the URL validity beforehand. But I just tested and timed and found it was not as slow as I thought, so I switched back to the simple try except.